sklearn/manifold/mds.py

```diff
@@ -72,8 +72,9 @@ def _smacof_single(dissimilarities, metric=True, n_components=2, init=None,
     n_samples = dissimilarities.shape[0]
     random_state = check_random_state(random_state)
 
-    sim_flat = ((1 - np.tri(n_samples)) * dissimilarities).ravel()
-    sim_flat_w = sim_flat[sim_flat != 0]
+    if not metric:
```

Comment:
why this change?

Reply:
My PR won't support non-metric MDS with masked arrays yet.  Since `sim_flat` is only used in the metric case, I changed it to only be computed in the metric case.